### PR TITLE
Add nonpip ci for arm

### PR DESCRIPTION
**Pull request by mseth10**
Adding a separate pipeline for running non pip tests for arm.

### CODE REVIEW

**Comment by mseth10** [comment]
@nitins17 can you take a look at this PR? Currently, there are many tests not building and many tests failing, which we need to investigate. But does the setup look good to you?

File: tensorflow/tools/ci_build/rel/ubuntu/cpu_arm64_nonpip.sh [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+export TF_NEED_HDFS=0
+export TF_NEED_OPENCL=0
+export TF_NEED_JEMALLOC=1
+export TF_NEED_VERBS=0
+export TF_NEED_AWS=0
+export TF_NEED_GDR=0
+export TF_NEED_OPENCL_SYCL=0
+export TF_NEED_COMPUTECPP=0
+export TF_NEED_KAFKA=0
+export TF_NEED_TENSORRT=0
```
Comment (nitins17) [inline tensorflow/tools/ci_build/rel/ubuntu/cpu_arm64_nonpip.sh:52]: I don't have familiarity with some of the settings that are used here. Could you let me know where you referenced these from?  I don't see any environment variables being set in the [build script](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/tools/ci_build/rel/ubuntu/cpu_py310_nonpip.sh) that we used to use for the non-pip tests.

**Review by nitins17 — CHANGES_REQUESTED** [review]

**Comment by nitins17** [comment]
Thanks so much for this! I have left my feedback above.

File: tensorflow/tools/ci_build/rel/ubuntu/cpu_arm64_nonpip.sh (hunk 1 above)
Comment (mseth10) [reply tensorflow/tools/ci_build/rel/ubuntu/cpu_arm64_nonpip.sh:52]: they are referenced from here: https://github.com/tensorflow/tensorflow/blob/db84fb33e9c075855685064dedf7923ee05b9eff/tensorflow/tools/ci_build/rel/ubuntu/cpu_arm64_pip.sh#L25

File: tensorflow/tools/ci_build/rel/ubuntu/cpu_arm64_nonpip.sh [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+# Export required variables for running pip_new.sh
+export OS_TYPE="UBUNTU"
+export CONTAINER_TYPE="CPU"
+
+# Get the default test targets for bazel.
+source tensorflow/tools/ci_build/build_scripts/DEFAULT_TEST_TARGETS.sh
+
+# Export optional variables for running pip_new.sh
+export TF_BUILD_FLAGS="--config=mkl_aarch64_threadpool --copt=-mtune=generic --copt=-march=armv8-a \
```
[the remark this answers is not in the document]
Comment (mseth10) [reply tensorflow/tools/ci_build/rel/ubuntu/cpu_arm64_nonpip.sh:62]: they are referenced from here: https://github.com/tensorflow/tensorflow/blob/db84fb33e9c075855685064dedf7923ee05b9eff/tensorflow/tools/ci_build/rel/ubuntu/cpu_arm64_pip.sh#L59-L60

**Comment by mseth10** [comment]
@elfringham can you please review the script for non-pip tests in file `tensorflow/tools/ci_build/rel/ubuntu/cpu_arm64_nonpip.sh`

**Comment by elfringham** [comment]
@mseth10 That seems mostly OK. You will need a list of tests to exclude, as you do in the cpu_arm64_pip.sh. I would suggest that you move the current list out of cpu_arm64_pip.sh into a new shell script that can then be sourced by both cpu_arm64_pip.sh and cpu_arm64_nonpip.sh into an environment variable. There will be additional tests that fail on the non-pip runs so there should be a second list, in the same new shell script, that is in another environment variable that can be used only by the non-pip run. That should stop any need to duplicate information and keep the maintenance of the exclusion list as simple as possible.

Also the non-pip CI runs so far have failed due to lack of disk space, so that will need to be resolved.

**Review by nitins17 — COMMENTED** [review]
LGTM, Thanks! I'll wait for the changes requested by @elfringham before approving.

**Comment by mseth10** [comment]
Addressed @elfringham 's comments. @nitins17 can you take another look?

**Review by nitins17 — APPROVED** [review]

**Comment by mseth10** [comment]
@nitins17 looks like it's missing CLA.. I checked and mine is present, looks like it's missing yours.

Also, after this PR is merged we need to enable more runners to share load. Do you have access to this page? https://github.com/tensorflow/tensorflow/settings/actions/runners/new?arch=arm64&os=linux

**Review by nitins17 — APPROVED** [review]

**Review by nitins17 — APPROVED** [review]

**Pull request merged**